### PR TITLE
feat: subgraph setup for the FleekNFA contracts

### PR DESCRIPTION
**Pull request by EmperorOrokuSaki**

### CODE REVIEW

File: deploy/verify_polyscan.js
Comment (zoruka) [inline deploy/verify_polyscan.js:1]: I think this file should be moved to `/scripts` and lets use the default naming patter for js files `verify-polyscan.js`

File: deploy/verify_polyscan.js
Comment (zoruka) [inline deploy/verify_polyscan.js:2]:
```suggestion
```
As this is supposed to be run within the hardhat environment, I think this is not needed

File: deploy/verify_polyscan.js
Comment (zoruka) [inline deploy/verify_polyscan.js:5]: Maybe we could get the contract address from deployments folder?

File: package.json
Comment (zoruka) [inline package.json:57]: Just leaving a comment to double check if `graphql` is a prod dependency or dev one.

File: subgraph/abis/FleekNFA.json
Comment (zoruka) [inline subgraph/abis/FleekNFA.json:1]: I think this can be got also from deployments folder.

File: subgraph/query-examples/main.ts
Comment (zoruka) [inline subgraph/query-examples/main.ts:26]: Just a hint, there is no changes needed here:
As this is just a function being called, we don't need to have a function is just use the module scope to run all. This would be cool if the `main` function was an async one, so the code could be rewriten like:
```suggestion
function async main() {
 const result = await execute(firstFiveMintsByUserDocument, {});
 console.log('\nFirst five mints by the user:');
 console.log(result.data?.transfers);

```

**Review by zoruka — COMMENTED** [review]
Just left some comments there. Great work! Waiting for the presentation 🚀

One thing to think about is about we having lots of projects under the same repo, maybe we could split apart to new repos instead of having a big monorepo. I'm also thinking about when we start creating the routines of the off chain functions. We already have contracts, ui and the subraph in the same repo @jsonsivar

File: deploy/verify_polyscan.js
Comment (EmperorOrokuSaki) [reply deploy/verify_polyscan.js:1]: local_deploy.js and mumbai_deploy.js both use underscores

Comment (EmperorOrokuSaki) [reply deploy/verify_polyscan.js:1]: moved the file to scripts.

File: subgraph/abis/FleekNFA.json
Comment (EmperorOrokuSaki) [reply subgraph/abis/FleekNFA.json:1]: It's autogenerated. The main abi file is in artifacts/contracts/FleekERC721.sol/FleekERC721.json but subgraph needs to store it separately.

File: subgraph/query-examples/main.ts
Comment (EmperorOrokuSaki) [reply subgraph/query-examples/main.ts:26]: Initially, that's how I wrote the script but I got Promise errors when I tried to run the script.

**Comment by csosamorales** [comment]
[sc-182]

**Comment by jsonsivar** [comment]
[sc-135]

**Comment by EmperorOrokuSaki** [comment]
[sc-265]

**Comment by EmperorOrokuSaki** [comment]
Note: 

We should wait for pr 82 to merge into this branch and then merge this to develop to avoid situations like before (when commit messages were duplicated)

This should not affect the reviews (I suggest we first review this and then review that pr)

File: subgraph/generated/FleekNFA/FleekNFA.ts
Comment (jsonsivar) [inline subgraph/generated/FleekNFA/FleekNFA.ts:19]: When does one use this file vs the other TS file in the query-examples folder?

File: subgraph/README.md
Comment (jsonsivar) [inline subgraph/README.md:53]: Maybe documenting the different schema and TS files would be useful in the readme

File: subgraph/query-examples/.graphclientrc.yml
Comment (jsonsivar) [inline subgraph/query-examples/.graphclientrc.yml:1]: Should we rename the folder to just `examples`? In the future if we have mutation examples or other things we can put it all under the examples folder.

**Review by jsonsivar — CHANGES_REQUESTED** [review]

**Comment by jsonsivar** [comment]
> Note:
> 
> We should wait for pr 82 to merge into this branch and then merge this to develop to avoid situations like before (when commit messages were duplicated)
> 
> This should not affect the reviews (I suggest we first review this and then review that pr)

I don't think so at all.  That PR is way to large already.  If anything we should cut down PR to make it smaller.  Even for a boilerplate/init type of PR I think this needs to be done in smaller pieces, especially since these are going into `develop` and not `main`.

I did an initial review of the PR and it was just around the structure and files etc.  I need to do another review still to dive deeper into the files but some of the questions in the first review will help with that.  Having gone through the structure, I think we should definitely try to make more incremental PRs and be very aware of what we are bringing to the repo.

I would in the future break this PR down into the following incremental PRs for review:
1. Initi setup PRIOR to generated ANY files. This will give a clear view of what you need to create in the first place and the "source" structure really.
2. Any generated stuff that's needed, some queries etc,
3. Examples
4. then add #82

File: subgraph/query-examples/.graphclientrc.yml
Comment (EmperorOrokuSaki) [reply subgraph/query-examples/.graphclientrc.yml:1]: That works. 👍 Inside I'll create a query folder, too.

File: subgraph/README.md
Comment (EmperorOrokuSaki) [reply subgraph/README.md:26]: No. Totally autogenerated by the graphclient, following their own documentation's "Querying" section.

File: subgraph/README.md
Comment (EmperorOrokuSaki) [reply subgraph/README.md:21]: That's what I did for the init command but it still created multiple versions of the abi.

File: subgraph/README.md
Comment (EmperorOrokuSaki) [reply subgraph/README.md:22]: All generated by graph cli. Also, I have not added any ABI to this subgraph folder (or any). They're generated by graph-cli as needed.

The build folder is what is uploaded to IPFS and then used for the subgraph deployment, so it makes sense to have the ABI here (even though it's really not in our hands to decide what it should contain or shouldn't since it is generated)

The schema and the yaml files are both based on the versions in the root of the subgraph directory, and also autogenerated.

The yaml and schema files at the root are written by me.

File: subgraph/README.md
Comment (EmperorOrokuSaki) [reply subgraph/README.md:23]: Yes.

File: subgraph/README.md
Comment (EmperorOrokuSaki) [reply subgraph/README.md:71]: The build command runs that so I don't think there is going to be a reason to run it separately. I'll add it just to be sure

File: subgraph/generated/FleekNFA/FleekNFA.ts
Comment (EmperorOrokuSaki) [reply subgraph/generated/FleekNFA/FleekNFA.ts:19]: This file is used in the src dir and query-examples is isolated from the other folders. The main.ts file in there is written by me.

File: subgraph/package.json
Comment (EmperorOrokuSaki) [reply subgraph/package.json:10]: Deploy should be removed since it uses subgraph studio (not available for hosted service)
I'll add description to docs for the rest.

**Comment by EmperorOrokuSaki** [comment]
> > Note:
> > 
> > We should wait for pr 82 to merge into this branch and then merge this to develop to avoid situations like before (when commit messages were duplicated)
> > 
> > This should not affect the reviews (I suggest we first review this and then review that pr)
> 
> I don't think so at all.  That PR is way to large already.  If anything we should cut down PR to make it smaller.  Even for a boilerplate/init type of PR I think this needs to be done in smaller pieces, especially since these are going into `develop` and not `main`.
> 
> I did an initial review of the PR and it was just around the structure and files etc.  I need to do another review still to dive deeper into the files but some of the questions in the first review will help with that.  Having gone through the structure, I think we should definitely try to make more incremental PRs and be very aware of what we are bringing to the repo.
> 
> I would in the future break this PR down into the following incremental PRs for review:
> 1. Initi setup PRIOR to generated ANY files. This will give a clear view of what you need to create in the first place and the "source" structure really.
> 2. Any generated stuff that's needed, some queries etc,
> 3. Examples
> 4. then add #82

The other PRs are dependant on the content of this PR. So they are based off of this branch. The solution to avoid duplicate commit messages was merging them into this first and then merging this.

Unless this pr was merged in develop and the others were based on the develop branch, we would still need to do this.

Will make sure to follow the steps for the next features.

File: subgraph/package.json
Comment (jsonsivar) [reply subgraph/package.json:10]: Just to summarize our chat, create is used to create the infra needed to run a subgraph and for loca we'd need that. But for testnet it's already deployed since it's a hosted version so therefore no create needed for testnet.

File: subgraph/README.md
Comment (jsonsivar) [reply subgraph/README.md:21]: Have discussion and then decide what to do.

File: subgraph/README.md
Comment (jsonsivar) [reply subgraph/README.md:22]: Maybe we could do this

- add to .gitignore (in this PR)
- in CI, do the build, and add some logs what we can (in another PR)

File: subgraph/README.md
Comment (jsonsivar) [reply subgraph/README.md:26]: Lets rename to examples, remove the client.

File: subgraph/generated/FleekNFA/FleekNFA.ts
Comment (jsonsivar) [reply subgraph/generated/FleekNFA/FleekNFA.ts:19]: Thank you for the explanation.  To summarize, the ones in the query-examples are generated by the graph client generator and are meant to be used by any clients of thegraph/this subgraph.  Meanwhile the ones in our generated folder is TS wrappers around the schema and events.

**Comment by EmperorOrokuSaki** [comment]
Applied the requested changes @jsonsivar

**Comment by jsonsivar** [comment]
> 

@EmperorOrokuSaki what about the graphql in the main package.json? https://github.com/fleekxyz/non-fungible-apps/pull/72#discussion_r1085894210

**Comment by EmperorOrokuSaki** [comment]
> > 
> 
> @EmperorOrokuSaki what about the graphql in the main package.json? [#72 (comment)](https://github.com/fleekxyz/non-fungible-apps/pull/72#discussion_r1085894210)

Works without. Pushed the commit to remove it.

File: subgraph/query-examples/main.ts
Comment (jsonsivar) [reply subgraph/query-examples/main.ts]: Let's quickly try to fix this and include but if its something major we can push to next sprint.

**Review by jsonsivar — APPROVED** [review]

**Pull request merged**
